pkg/functionconfig/types.go

```diff
@@ -34,18 +34,30 @@ type DataBinding struct {
 	Attributes map[string]string `json:"attributes,omitempty"`
 }
 
+// Partition is a partition information
+type Partition struct {
+	ID         string  `json:"id"`
+	Checkpoint *string `json:"checkpoint,omitempty"`
+}
+
 // Trigger holds configuration for a trigger
 type Trigger struct {
-	Class         string                 `json:"class"`
-	Kind          string                 `json:"kind"`
-	Disabled      bool                   `json:"disabled,omitempty"`
-	MaxWorkers    int                    `json:"maxWorkers,omitempty"`
-	URL           string                 `json:"url,omitempty"`
-	Paths         []string               `json:"paths,omitempty"`
-	NumPartitions int                    `json:"numPartitions,omitempty"`
-	User          string                 `json:"user,omitempty"`
-	Secret        string                 `json:"secret,omitempty"`
-	Attributes    map[string]interface{} `json:"attributes,omitempty"`
+	Class      string      `json:"class"`
+	Kind       string      `json:"kind"`
+	Disabled   bool        `json:"disabled,omitempty"`
+	MaxWorkers int         `json:"maxWorkers,omitempty"`
+	URL        string      `json:"url,omitempty"`
+	Paths      []string    `json:"paths,omitempty"`
+	User       string      `json:"user,omitempty"`
+	Secret     string      `json:"secret,omitempty"`
+	Partitions []Partition `json:"partitions,omitempty"`
+
+	// Dealer Information
```

Comment:
Why is this required?

Reply:
@yaronha Asked for this to be at top level of configuration.

Reply:
@yaronha This is for processor who starts without a dealer.